### PR TITLE
Updating Vector<T> to support opt-in 512-bit widths

### PR DESCRIPTION
**Pull request by tannergooding**

### CODE REVIEW

**Comment by stephentoub** [comment]
Do we have any places in our own use of `Vector<T>` where we take a dependency on it not being larger than 256?  I know we have one here:
https://github.com/dotnet/runtime/blob/95bf3d9922535ce2975c21548e049e94b21a52cb/src/libraries/System.Linq/src/System/Linq/Range.SpeedOpt.cs#L45-L48
but we explicitly guard it on Count because we wrote it knowing this might be coming (though we could also update that now to construct the Vector with a larger sequence so it lights up on 512.)  I'm wondering if we might have any others that would need to be tweaked.

**Comment by tannergooding** [comment]
> but we explicitly guard it on Count because we wrote it knowing this might be coming (though we could also update that now to construct the Vector with a larger sequence so it lights up on 512.) I'm wondering if we might have any others that would need to be tweaked.

I know we have a few places throughout the BCL which guard against it being larger than 256-bits and will use an alternative path if it is. Those will definitely need to be found and updated where relevant, but shouldn't be blocked on this PR (especially since its opt-in).

Some of the patterns like `new Vector<int>((ReadOnlySpan<int>)[0, 1, 2, 3, 4, 5, 6, 7]); ` are ones  I'd like to "solve" by exposing new `Create` APIs. Something like `CreateSequence(int start, int step)` (but with better naming). That should help avoid creating unnecessarily large RVA statics and make the code more portable.

File: src/coreclr/jit/simdashwintrinsiclistarm64.h [hunk 1]
```diff
@@ -202,99 +202,99 @@ SIMD_AS_HWINTRINSIC_ID(Vector4,     WithElement,
...
 //  Vector<T> Intrinsics
-SIMD_AS_HWINTRINSIC_ID(VectorT128,  Abs,                                                    1,         {NI_VectorT128_Abs,                             NI_VectorT128_Abs,                              NI_VectorT128_Abs,                              NI_VectorT128_Abs,                              NI_VectorT128_Abs,                              NI_VectorT128_Abs,                              NI_VectorT128_Abs,                              NI_VectorT128_Abs,                              NI_VectorT128_Abs,                              NI_VectorT128_Abs},                             SimdAsHWIntrinsicFlag::None)
```
Comment (kunalspathak) [inline src/coreclr/jit/simdashwintrinsiclistarm64.h:205]: the changes in this and `simdashwintrinsiclistxarch.h` are mainly renaming `VectorT128` to `VectorT`?

File: src/coreclr/jit/simdashwintrinsic.cpp [hunk 2]
```diff
@@ -682,30 +641,11 @@ GenTree* Compiler::impSimdAsHWIntrinsicSpecial(NamedIntrinsic       intrinsic,
             break;
         }
 
-        case NI_VectorT128_Sum:
-        {
-            if (varTypeIsFloating(simdBaseType))
```
Comment (kunalspathak) [inline src/coreclr/jit/simdashwintrinsic.cpp:687]: I see that the case for `VectorT_Sum` is moved below. wondering we don't need `InstructionSet_*` checks anymore?

**Review by kunalspathak — APPROVED** [review]
LGTM. Thanks. The arm changes will come handy for SVE work, right?

File: src/coreclr/jit/simdashwintrinsiclistarm64.h (hunk 1 above)
Comment (tannergooding) [reply src/coreclr/jit/simdashwintrinsiclistarm64.h:205]: Right, that's the only change here which should simplify adding future sizes to `Vector<T>` as we no longer have to duplicate table entries per size.

File: src/coreclr/jit/simdashwintrinsic.cpp (hunk 2 above)
Comment (tannergooding) [reply src/coreclr/jit/simdashwintrinsic.cpp:687]: The implementation was updated to not use the horizontal operations anymore, as it has better performance on modern hardware and was needed for Vector512_Sum anyways.

**Pull request merged**